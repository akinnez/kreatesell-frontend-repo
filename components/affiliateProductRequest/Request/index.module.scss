@use "public/css/mixins";
@use "public/css/screens";

.form__wrapper {
  max-width: 645px;

  div[class~="ant-form-item-label"] {
    padding-bottom: 0;
  }
}

.text {
  margin-bottom: 20px;
}

.submit__btn {
  border-radius: 8px;
}

@include screens.screen("medium-large") {
  .form__wrapper {
    padding: 24px 24px;
    background-color: #fff;
    box-shadow: 0px 20px 100px rgba(36, 53, 72, 0.05);
    border-radius: 8px;
  }
}
