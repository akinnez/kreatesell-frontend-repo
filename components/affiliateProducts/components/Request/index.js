import { useState } from "react";
import { useRouter } from "next/router";
import { Button, Form, Input, Typography } from "antd";
import { Formik } from "formik";
import RequestSuccessModal from "../RequestSuccessModal";
import { showToast } from "utils";
import axiosApi from "utils/axios";
import { AffiliatePermission } from "validation/AffiliatePermission.validation";
import styles from "./index.module.scss";

const { TextArea } = Input;

const Request = () => {
  const [showModal, setShowModal] = useState(false);
  const [hasSubmitted, setHasSubmitted] = useState(false);
  const router = useRouter();

  const submitHandler = (values, actions) => {
    const data = {
      requested_product_id: router.query.pId,
      note: values.permission.trim(),
    };

    axiosApi.request(
      "post",
      `${process.env.BASE_URL}affiliate/create-product-marketing-request`,
      () => {
        setShowModal(true);
        actions.setSubmitting(false);
      },
      err => {
        showToast(err, "error");
        actions.setSubmitting(false);
      },
      data
    );
  };

  const handleHideModal = () => {
    setShowModal(false);
    setHasSubmitted(true);
  };

  return (
    <>
      <div className={styles.form__wrapper}>
        <Formik
          initialValues={{
            permission: "",
          }}
          validationSchema={AffiliatePermission}
          onSubmit={submitHandler}
        >
          {formik => (
            <Form layout="vertical" onFinish={formik.handleSubmit} size="large">
              <Form.Item
                name="permission"
                label="Request Permission"
                validateStatus={
                  formik.touched.permission &&
                  formik.errors.permission &&
                  "error"
                }
                help={formik.touched.permission && formik.errors.permission}
              >
                <TextArea
                  rows={5}
                  placeholder="Fill out how you want to promote this product"
                  {...formik.getFieldProps("permission")}
                  disabled={hasSubmitted}
                />
              </Form.Item>
              <div className={styles.text}>
                <p>
                  <Typography.Text>
                    Almost there, Click the next button to continue
                  </Typography.Text>
                </p>
              </div>
              <Form.Item>
                {hasSubmitted ? (
                  <Button
                    type="primary"
                    className={styles.submit__btn}
                    disabled
                  >
                    Request Approval
                  </Button>
                ) : (
                  <Button
                    htmlType="submit"
                    type="primary"
                    loading={formik.isSubmitting}
                    className={styles.submit__btn}
                  >
                    Request Approval
                  </Button>
                )}
              </Form.Item>
            </Form>
          )}
        </Formik>
      </div>
      {showModal && (
        <RequestSuccessModal
          showModal={showModal}
          handleHideModal={handleHideModal}
        />
      )}
    </>
  );
};

export default Request;
