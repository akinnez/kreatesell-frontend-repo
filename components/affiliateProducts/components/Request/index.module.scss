@use "public/css/colors";
@use "public/css/mixins";
@use "public/css/screens";

.form__wrapper {
  max-width: 645px;

  form[class^="ant-form"] textarea {
    line-height: 30px;
    font-size: 14px;
    overflow: hidden;

    &[disabled] {
      background-color: colors.$white-200;
      color: colors.$gray-200;
    }
  }

  div[class$="ant-form-item-label"] {
    padding: 0;
    font-weight: 500;
    font-size: 16px;
    color: colors.$black-100;
  }
}

.text {
  margin-top: 25px;
  margin-bottom: 20px;

  p {
    margin-bottom: 0;

    & > [class="ant-typography"] {
      font-weight: 500;
      font-size: 16px;
      color: colors.$base-gray;
    }
  }
}

.submit__btn {
  border-radius: 8px;
  font-weight: 700;
  font-size: 14px;

  &[disabled] {
    background-color: colors.$gray-700 !important;
    color: colors.$base-white;

    &:hover {
      color: colors.$base-white;
    }
  }
}

.content {
  margin-bottom: 40px;
  text-align: center;

  p {
    margin-bottom: 0;

    &:first-child > span[class="ant-typography"] {
      font-size: 30px;
      font-weight: 600;
      color: colors.$black-100;
    }

    &:last-child > span[class="ant-typography"] {
      font-size: 16px;
      font-weight: 500;
      color: colors.$gray-200;
    }
  }
}

@include screens.screen("medium-large") {
  .form__wrapper {
    padding: 24px 24px;
    background-color: #fff;
    box-shadow: 0px 20px 100px rgba(36, 53, 72, 0.05);
    border-radius: 8px;
  }
}
