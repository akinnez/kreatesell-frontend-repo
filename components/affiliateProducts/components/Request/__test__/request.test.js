import renderer from "react-test-renderer";
import Request from "..";

Object.defineProperty(window, "matchMedia", {
  writable: true,
  value: jest.fn().mockImplementation((query) => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(),
    removeListener: jest.fn(),
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});

it("renders a snapshot of <Request />", () => {
  const request = renderer.create(<Request />).toJSON();
  expect(request).toMatchSnapshot();
});
