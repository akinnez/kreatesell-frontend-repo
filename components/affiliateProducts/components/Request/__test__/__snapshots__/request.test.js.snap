// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot of <Request /> 1`] = `
<div
  className="form__wrapper"
>
  <form
    className="ant-form ant-form-vertical ant-form-large"
    onReset={[Function]}
    onSubmit={[Function]}
  >
    <div
      className="ant-row ant-form-item"
      style={Object {}}
    >
      <div
        className="ant-col ant-form-item-label"
        style={Object {}}
      >
        <label
          className=""
          htmlFor="permission"
          title="Request Permission"
        >
          Request Permission
        </label>
      </div>
      <div
        className="ant-col ant-form-item-control"
        style={Object {}}
      >
        <div
          className="ant-form-item-control-input"
        >
          <div
            className="ant-form-item-control-input-content"
          >
            <textarea
              className="ant-input ant-input-lg"
              id="permission"
              onChange={[Function]}
              onCompositionEnd={[Function]}
              onCompositionStart={[Function]}
              onKeyDown={[Function]}
              placeholder="Fill out how you want to promote this product"
              rows={5}
              style={Object {}}
              value=""
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="text"
    >
      <p>
        <span
          className="ant-typography"
          onClick={null}
          style={
            Object {
              "WebkitLineClamp": undefined,
            }
          }
        >
          Almost there, Click the next button to continue
        </span>
      </p>
    </div>
    <div
      className="ant-row ant-form-item"
      style={Object {}}
    >
      <div
        className="ant-col ant-form-item-control"
        style={Object {}}
      >
        <div
          className="ant-form-item-control-input"
        >
          <div
            className="ant-form-item-control-input-content"
          >
            <button
              className="ant-btn ant-btn-primary ant-btn-lg submit__btn"
              onClick={[Function]}
              type="submit"
            >
              <span>
                Request Approval
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </form>
</div>
`;
